packages/dd-trace/src/profiling/config.js

```diff
@@ -191,6 +193,44 @@ class Config {
     logExperimentalVarDeprecation('CPU_ENABLED')
     checkOptionWithSamplingContextAllowed(this.cpuProfilingEnabled, 'CPU profiling')
 
+    const uploadCompression0 = coalesce(options.uploadCompression, DD_PROFILING_DEBUG_UPLOAD_COMPRESSION, 'on')
+    let [uploadCompression, level0] = uploadCompression0.split('-')
+    if (!['on', 'off', 'gzip', 'zstd'].includes(uploadCompression)) {
+      logger.warn(`Invalid profile upload compression method "${uploadCompression0}". Will use "on".`)
+      uploadCompression = 'on'
+    } else if (uploadCompression === 'zstd' && !satisfies(process.versions.node, '>=23.8.0')) {
+      logger.warn(
+        'Profile upload compression method "zstd" is only supported on Node.js 23.8.0 or above. Will use "on".')
+    }
+    let level = level0 ? parseInt(level0, 10) : undefined
+    if (level !== undefined) {
+      if (['on', 'off'].includes(uploadCompression)) {
+        logger.warn(`Compression levels are not supported for "${uploadCompression}".`)
+        level = undefined
+      } else if (isNaN(level)) {
+        logger.warn(
+          `Invalid compression level "${level0}". Will use default level.`)
+        level = undefined
+      } else if (level < 0) {
+        // Best I can tell this can't happen due to splitting the env var at
+        // minus sign, but let's be exhaustive.
+        logger.warn(`Invalid compression level ${level}. Will use 0.`)
+        level = 0
+      } else if (level > 9) {
+        logger.warn(`Invalid compression level ${level}. Will use 9.`)
+        level = 9
+      } else {
+        level |= 0
```

Comment:
Not sure what is the intent here ?

Reply:
Levels are supposed to be integers. I figured we can silently turn `zstd-3.14` into `zstd-3`. But we can also warn about it?

Reply:
hum, `level` is the result of `parseInt(level0, 10)` so it should be an integer here

Reply:
aaah, of course!